AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/ui/activity/dashboard/DeepLinkListActivity.java

```diff
@@ -0,0 +1,121 @@
+package com.example.appactionvisualizer.ui.activity.dashboard;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ExpandableListAdapter;
+import android.widget.ExpandableListView;
+
+import com.example.appactionvisualizer.R;
+import com.example.appactionvisualizer.databean.ActionType;
+import com.example.appactionvisualizer.databean.AppActionProtos;
+import com.example.appactionvisualizer.databean.AppActionProtos.Action;
+import com.example.appactionvisualizer.databean.AppActionProtos.AppAction;
+import com.example.appactionvisualizer.databean.AppActionProtos.FulfillmentOption;
+import com.example.appactionvisualizer.databean.AppActionsGenerator;
+import com.example.appactionvisualizer.databean.Tuple;
+import com.example.appactionvisualizer.ui.activity.CustomActivity;
+import com.example.appactionvisualizer.ui.adapter.ExpandableAdapter;
+import com.example.appactionvisualizer.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+// Display deep links using an expandable list view
+public class DeepLinkListActivity extends CustomActivity {
+  // For each action name, we need a tuple of <AppAction, Action, FulfillmentOption> data
+  // so that the link can jump into ParameterActivity and parse required data to activity.
+  Map<String, List<Tuple<AppAction, Action, FulfillmentOption>>> intentMap;
```

Comment:
I think tuple is meaningful to me since I've used this in c++, I'm a bit confused that Java didn't provide similar class---a collection of heterogeneous values. Maybe I could use List<Object> but I don't think that's reader-friendly.
What would you suggest as the name?

Reply:
maybe do something like
Class Fulfillment {
   AppAction appAction;
   Action action;
   FulfillmentOption fultillmentOption;
}
??
In Java, we usually use a class to represent the combo like this.

Reply:
That sounds reasonable, will fix it